### ISSUE TITLE
Dropdown menu items should have role="option"

### ISSUE THREAD

**Issue by enricoberti** [carbon-design-system/carbon#5368]
## Environment

> MacOS

> Chrome, FF

> Storybook's Accessibility Tab

## Detailed description

Testing on master, the original issue was open on [carbon-addons-iot-react](https://github.com/ibm/carbon-addons-iot-react/issues/616).

It's marked as 'critical', with the info given that
- Certain ARIA roles must contain particular children
- Required ARIA child role not present: option

## Steps to reproduce the issue

Verify on https://react.carbondesignsystem.com/?path=/story/dropdown--default that `role="option"` is missing from the dropdown options

## Additional information

Possibly a dupe of #5211

**Comment by enricoberti** [comment]
Will be closed with #5373
